File: Cards/Soldier.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__ClassExtends = ____lualib.__TS__ClassExtends
local ____exports = {}
local ____Enums = require("Enums")
local Ranks = ____Enums.Ranks
local ____Card = require("Cards.Card")
local Card = ____Card.default
____exports.default = __TS__Class()
local Soldier = ____exports.default
Soldier.name = "Soldier"
__TS__ClassExtends(Soldier, Card)
function Soldier.prototype.____constructor(self, gameManager, suit)
    Card.prototype.____constructor(
        self,
        gameManager,
        suit,
        Ranks.SOLDIER,
        3,
        0,
        "Soldier"
    )
end
return ____exports

File: Cards/Soldier.ts
import { Ranks, Suits } from "../Enums";
import GameManager from "../GameManager";
import Card from "./Card";

export default class Soldier extends Card {
  constructor(gameManager: GameManager, suit: Suits) {
    super(
      gameManager,
      suit,
      Ranks.SOLDIER,
      3,
      0,
      "Soldier",
      "Assets/Images/SoldierRank.png",
      {
        edelName: "Knight",
        edelPower: 7,
        edelValue: 0,
        edelRankAssetPath: "Assets/Images/KnightRank.png",
      }
    );
  }
}
